src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -330,6 +335,22 @@ public abstract class AbstractCompilerMojo extends AbstractMojo {
     @Parameter
     private List<DependencyCoordinate> annotationProcessorPaths;
 
+    /**
+     * <p>
+     * Whether to use the Maven dependency management section when resolving transitive dependencies of annotation
+     * processor paths.
+     * </p>
+     * <p>
+     * This flag does not enable / disable the ability to get the version of annotation processor paths
+     * (the top-level paths) from dependency management section. It only influences the resolution of
+     * transitive dependencies of those top-level paths.
+     * </p>
+     *
+     * @since 3.12.0
+     */
+    @Parameter(defaultValue = "false")
```

Comment:
:beer: for this: new feature, if alters behavior, should be disabled by def

Reply:
It may definitely change the resulting `processorpath`, thus changing the behavior. So it should be opt-in.